a10_neutron_lbaas/v2/handler_base_v2.py

```diff
@@ -15,13 +15,38 @@
 import a10_neutron_lbaas.handler_base as base
 import neutron_ops
 
+import re
 
 class HandlerBaseV2(base.HandlerBase):
-
     def __init__(self, a10_driver, openstack_manager, neutron=None):
         super(HandlerBaseV2, self).__init__(a10_driver)
         self.openstack_manager = openstack_manager
         if neutron:
             self.neutron = neutron
         else:
             self.neutron = neutron_ops.NeutronOpsV2(self)
+
+    """
+    Pass in an element, it's openstack name, and a dictionary of matches.
+    """
+    def _get_name_matches(self, elem, os_name, redict):
+        # for each key in the vport_defaults dictionary
+        for k, v in redict.iteritems():
+            # check to see if the regex value matches.
+            v = redict[k]
+
+            regex_str = v["regex"]
+            json_merge = v["json"]
+            regex = re.compile(regex_str)
+            matched = regex.search(os_name)
```

Comment:
This will currently cause objects (e.g. listeners) to fail if a name isn't set.

e.g.
```

Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/neutron_lbaas/services/loadbalancer/plugin.py", line 486, in _call_driver_operation
    driver_method(context, db_entity)
  File "/usr/lib/python2.7/site-packages/neutron_lbaas/drivers/a10networks/driver_v2.py", line 379, in create
    self.driver.a10.listener.create(context, listener)
  File "/usr/lib/python2.7/site-packages/a10_neutron_lbaas/v2/handler_listener.py", line 264, in create
    self._create(c, context, listener)
  File "/usr/lib/python2.7/site-packages/a10_neutron_lbaas/v2/handler_listener.py", line 260, in _create
    c, context, listener)
  File "/usr/lib/python2.7/site-packages/a10_neutron_lbaas/v2/handler_listener.py", line 153, in _set
    template_args.update(**self._get_vport_defaults(c, os_name))
  File "/usr/lib/python2.7/site-packages/a10_neutron_lbaas/v2/handler_listener.py", line 333, in _get_vport_defaults
    self._get_name_matches(rv, vport_name, self._get_expressions(c))
  File "/usr/lib/python2.7/site-packages/a10_neutron_lbaas/v2/handler_base_v2.py", line 41, in _get_name_matches
    matched = regex.search(os_name)
TypeError: expected string or buffer
```